packages/frontend/src/helpers/variables.ts

```diff
@@ -113,26 +113,28 @@ const process = (data: any, parentKey?: any, index?: number): RawVariable[] => {
   })
 }
 
-export function extractVariables(steps: IStep[]): StepWithVariables[] {
-  if (!steps) {
+export function extractVariables(
+  executionSteps: IExecutionStep[],
+): StepWithVariables[] {
+  if (!executionSteps) {
     return []
   }
-  return steps
-    .filter((step: IStep) => {
-      const hasExecutionSteps = !!step.executionSteps?.length
-
-      return hasExecutionSteps
+  return executionSteps
+    .filter((executionStep: IExecutionStep) => {
+      const hasDataOut = Object.keys(executionStep.dataOut ?? {}).length
+      return hasDataOut
     })
-    .map((step: IStep, index: number) => ({
-      id: step.id,
+    .map((executionStep: IExecutionStep, index: number) => ({
+      id: executionStep.id,
```

Comment:
can i check what's the reason for changing from `step.id` to the `executionStep.id` ah